src/components/application_manager/test/request_controller/request_controller_test.cc

```diff
@@ -211,40 +216,18 @@ TEST_F(RequestControllerTestClass,
   // app_hmi_level_none_time_scale_max_requests_ equals 0
   // (in the default settings they setted to 0)
   for (size_t i = 0; i < kMaxRequestAmount; ++i) {
+    RequestPtr request_valid = GetMockRequest(i);
+    EXPECT_CALL(*request_valid, GetApplicationManager())
+        .WillRepeatedly(ReturnRef(app_mngr_));
+    EXPECT_CALL(app_mngr_, IsStopping()).WillRepeatedly(Return(false));
     EXPECT_EQ(RequestController::TResult::SUCCESS,
               AddRequest(default_settings_,
-                         GetMockRequest(i),
+                         request_valid,
                          RequestInfo::RequestType::MobileRequest,
                          mobile_apis::HMILevel::HMI_FULL));
   }
 }
 
-TEST_F(
-    RequestControllerTestClass,
-    CheckPosibilitytoAdd_ExcessPendingRequestsAmount_TooManyPendingRequests) {
-  TestSettings settings;
-  settings.pending_requests_amount_ = kNumberOfRequests;
-
-  request_ctrl_->DestroyThreadpool();
-
-  // Adding requests to fit in pending_requests_amount_
-  for (size_t i = 0; i < kNumberOfRequests; ++i) {
-    EXPECT_EQ(RequestController::TResult::SUCCESS,
-              AddRequest(settings,
-                         GetMockRequest(),
-                         RequestInfo::RequestType::MobileRequest,
-                         mobile_apis::HMILevel::HMI_FULL));
-  }
-
-  // Trying to add one more extra request
-  // Expect overflow and TOO_MANY_PENDING_REQUESTS result
-  EXPECT_EQ(RequestController::TResult::TOO_MANY_PENDING_REQUESTS,
-            AddRequest(settings,
-                       GetMockRequest(),
-                       RequestInfo::RequestType::MobileRequest,
-                       mobile_apis::HMILevel::HMI_FULL));
-}
-
```

Comment:
Do we know why this test was removed?

Reply:
That test seems to have been removed due to the change in this line https://github.com/smartdevicelink/sdl_core/pull/3939/files#diff-4f3d0bb78ef3a8c3c3a38cc38dbe26e370440ef9c134c2054f7c0b98b1dd1501R208. The test used request_ctrl_->DestroyThreadpool(); and, due to the added TPoolState::STOPPED != pool_state_  check, no pending requests would be added to the mobile_request_list_ so the number of pending requests wouldn't increase at all (TOO_MANY_PENDING_REQUESTS wouldn't be triggered)